grakn-core/src/main/java/ai/grakn/concept/ConceptId.java

```diff
@@ -59,15 +60,22 @@ public Object getRawValue(){
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-
-        ConceptId conceptId1 = (ConceptId) o;
-
-        return getValue().equals(conceptId1.getValue());
+        ConceptId cast = (ConceptId) o;
+        if (conceptId instanceof String) {
```

Comment:
I don't understand why you need this `if else`. For a given graph (for example a titan graph) the concept Ids will always be strings. Whats the need for this ? Why not just toString both and compare? Is the toString that expensive?